index.html

```diff
@@ -204,7 +304,73 @@
 						bindKey: { win: 'Ctrl-Enter', mac: 'Command-Enter' },
 						exec: vm.run_code,
 					})
+
+					// Control-S -> Save file
+					vm.editor.commands.addCommand({
+						name: 'save_code',
+						bindKey: { win: 'Ctrl-S', mac: 'Ctrl-S' },
+						exec: vm.save_code,
+					})
+				},
+
+				save_code() {
+					const vm = this
+
+					// Get script name
+					var script_name = prompt("Enter script name:")
+
+					// If user hits the cancel button
+					if (script_name == null){
+						return
+					}
+
+					if (script_name.length > 0){
```

Comment:
In general, do error handling up front, like you did with `if (script_name == null){`
```js
if (script_name.length == 0) {
  alert("Script name can't be empty");
  return;
}
```

Reply:
updated with fix